### PR TITLE
open source compliance: copyright headers

### PR DESCRIPTION
**Pull request by GhiathAjam**
1. Add copyright headers to all applicable files 
3. ~Reivew license.txt files for potential updates~
4. ~Look into other requirements involving (README, CODE_OF_CONDUCT, CONTRIBUTING, SECURITY, ... etc]~

~All 3 points are still WIP. Deployed for fast feedback~

Edit: will limit to pt. 1 only. Title edited.

### CODE REVIEW

**Comment by albert17** [comment]
Did you look at [copyright guidance](https://microsoft.sharepoint.com/sites/CELAWeb-Copyrights-Trademarks-And-Patents/SitePages/copyrights-legal-notices.aspx?xsdata=MDV8MDJ8fDkwMGFiYzM5M2IyZjRhODMwZWM0MDhkYzJlNjA0ZDgxfDcyZjk4OGJmODZmMTQxYWY5MWFiMmQ3Y2QwMTFkYjQ3fDB8MHw2Mzg0MzYyMzgzMzE4MDk0NzZ8VW5rbm93bnxWR1ZoYlhOVFpXTjFjbWwwZVZObGNuWnBZMlY4ZXlKV0lqb2lNQzR3TGpBd01EQWlMQ0pRSWpvaVYybHVNeklpTENKQlRpSTZJazkwYUdWeUlpd2lWMVFpT2pFeGZRPT18MXxMM1JsWVcxekx6RTVPbXhpUm1KMGRrbDZSM0JvT0dWc05WaDFiMjQ1V1ZGMlIxWkhZMHBXYUdZNFlWVjRUbVpvVTBoSlNra3hRSFJvY21WaFpDNTBZV04yTWk5amFHRnVibVZzY3k4eE9UcHNZa1ppZEhaSmVrZHdhRGhsYkRWWWRXOXVPVmxSZGtkV1IyTktWbWhtT0dGVmVFNW1hRk5JU1VwSk1VQjBhSEpsWVdRdWRHRmpkakl2YldWemMyRm5aWE12TVRjd09EQXlOekF6TVRnNE13PT18YTBkMzVmZDkxODVhNGQzMWFjOGYwOGRjMmU2MDRkN2V8NDFlMjM3Y2UwMGE0NGU2ZDg1NGM5NmNiMWUwNDgxNzA%3D&sdata=WFg5OXUyakF6Zml4SWhzR3Nqc04rbXdtdkF6Zit1M2tIbjFHT0hKeFdSOD0%3D&ovuser=72f988bf-86f1-41af-91ab-2d7cd011db47%2Caalvarez%40microsoft.com&OR=Teams-HL&CT=1720616914284&clickparams=eyJBcHBOYW1lIjoiVGVhbXMtRGVza3RvcCIsIkFwcFZlcnNpb24iOiI0OS8yNDA2MTMxODQwOCJ9) ?

**Comment by albert17** [comment]
No sure exactly what you are doing in 2 and 3, but do 1 (copyright headers) in a specific PR please

File: .github/workflows/black.yml [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# Copyright (c) Microsoft Corporation.
```
Comment (albert17) [inline .github/workflows/black.yml:1]: Could you enforce headers in CI/Workflows?

Comment (albert17) [inline .github/workflows/black.yml:1]: No sure exactly what you are doing in points 2) and 3) in description, but do 1 (copyright headers) in a specific PR please

**Review by nader-m-elhadedy — APPROVED** [review]

File: .github/workflows/black.yml (hunk 1 above)
Comment (GhiathAjam) [reply .github/workflows/black.yml:1]: I'm not sure what you mean but yaml files do support comments yes, and the workflow ran and passed no problems (check youself below)

Comment (GhiathAjam) [reply .github/workflows/black.yml:1]: Alright limited this PR to (1). Thanks.

**Comment by GhiathAjam** [comment]
@albert17 kindly check the updated description and indicate any recommended changes prior to merging. Thanks

File: .github/workflows/black.yml (hunk 1 above)
Comment (albert17) [reply .github/workflows/black.yml:1]: I meant. What if I create a new file with no headers? Will the lint workflow fail?

**Review by albert17 — APPROVED** [review]
Adding headers LGTM, but we should check in linter. It is fine in other PR

**Comment by GhiathAjam** [comment]
@albert17 This is a very good idea I think it should be implemented company-wise! (maybe even expand to other open-source compliance rules !).
Thanks Alberto!

File: .github/workflows/black.yml (hunk 1 above)
Comment (GhiathAjam) [reply .github/workflows/black.yml:1]: Good idea. I believe it's possible and I do like it. Thanks.

**Pull request merged**